### ISSUE TITLE
Undo/Redo: Root context incorrectly marked as pending

### ISSUE THREAD

**Issue by raineorshine** [cybersemics/em#1344]
@anmolarora1:

> However, I've found another issue that needs to be solved to fully fix #1339.
> The current implementation is leading us into an invalid state, due to incorrect values of root context (falsely marked as pending)
> 
> [In our initialState](https://github.com/cybersemics/em/blob/4615ad614a7e853f61aa9f13c2f67c8c150a55b6/src/util/initialState.ts#L21), we're marking the root thought as pending and the `pending` status is updated through [updateThoughts](https://github.com/cybersemics/em/blob/4615ad614a7e853f61aa9f13c2f67c8c150a55b6/src/action-creators/pull.ts#L45) to false only after the children get pulled. But when there are no children, the status doesn't change, leading to an erroneous state - It's the exact state that we reach after hitting undo when there's a single thought under the Root, as described in the [original issue](https://github.com/cybersemics/em/issues/1339).
> That's my finding so far, and I'd like to hear your thoughts before presenting a solution to this one. It might be causing other issues, but we can't say for sure since it's a rare state.

**Comment by raineorshine** [comment]
Good catch. We should be able to set the root `pending` status to false once we know there are no children. It is similar to how we set `isLoading` to false in `pull` when the root has been pulled:

https://github.com/cybersemics/em/blob/4615ad614a7e853f61aa9f13c2f67c8c150a55b6/src/action-creators/pull.ts#L46-L54

I made a first attempt in #1345. Will you review? It seems to be helping.
